### PR TITLE
Initial setup of delivery_sim and package_calculation + unit tests

### PR DESCRIPTION
**Pull request by oliverdalgaard**

### CODE REVIEW

**Review by xahen — APPROVED** [review]
catblunt

File: src/resource_calculation.c [hunk 1]
```diff
@@ -2,4 +2,32 @@
 #include <stdlib.h>
 
 #include "resource_calculation.h"
-#include "package_distribution.h"
\ No newline at end of file
+#include "delivery_sim.h"
+
+// This can be a different value. Should we not just calculate it in calculate_trucks()?
+#define truck_volume 41.472
```
Comment (oliverdalgaard) [inline src/resource_calculation.c]:
```suggestion
#define TRUCK_VOLUME 41.472
```

File: src/resource_calculation.c [hunk 2]
```diff
@@ -2,4 +2,32 @@
...
+
+// Maybe we should take weight into account as well.
+void calculate_trucks(package_t package, double *volume_filled, int *truck_amount) {
+    double volume = package.height * package.width * package.length;
+
+    if (*volume_filled + volume < truck_volume) {
```
Comment (oliverdalgaard) [inline src/resource_calculation.c]:
```suggestion
    if (*volume_filled + volume < TRUCK_VOLUME) {
```

File: src/run_program.c [hunk 3]
```diff
@@ -1,9 +1,16 @@
+#include <delivery_sim.h>
+#include <package_distribution.h>
+#include <resource_calculation.h>
```
Comment (oliverdalgaard) [inline src/run_program.c]:
```suggestion
```

**Comment by xahen** [comment]
catblunt

**Comment by xahen** [comment]
catblunt

**Comment by xahen** [comment]
catblunt

**Comment by xahen** [comment]
catblunt

File: test/test_functions.c [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+    for (int i = 0; i < 20; i++) {
+        trucks_needed += calculate_trucks(package, &volume_filled, 0);
+    }
+
+    CHECK_EQ_INT(trucks_needed, 3);
+})
+
+// TODO:
+// TEST PRINT FUNCTIONS
+// THIS IS NOT POSSIBLE WITHOUT RETURNING A STRING (THESE FUNCTIONS DO NOT NEED TESTS ANYWAY)
```
Comment (oliverdalgaard) [inline test/test_functions.c:103]:
```suggestion
```

**Review by kresten-j — APPROVED** [review]

**Comment by xahen** [comment]
catblunt

**Pull request merged**